src/dos/drive_virtual.cpp

```diff
@@ -427,51 +425,34 @@ bool Virtual_Drive::MakeDir(char * /*dir*/) {
 bool Virtual_Drive::TestDir(char * dir) {
 	assert(dir);
 	if (!dir[0]) return true;		//only valid dir is the empty dir
-	auto cur_file = first_file;
-	while (cur_file) {
-		if (cur_file->isdir && iequals(cur_file->name, dir)) {
-			return true;
-		}
-		cur_file = cur_file->next;
+
+	if (find_file_by_predicate(first_file, [dir](vfile_block_t cur_file) {
+		    return cur_file->isdir && iequals(cur_file->name, dir);
+	    })) {
+		return true;
 	}
 	return false;
 }
 
 bool Virtual_Drive::FileStat(const char* name, FileStat_Block * const stat_block){
 	assert(name);
-	auto cur_file = first_file;
-	while (cur_file) {
-		unsigned int onpos = cur_file->onpos;
-		if (strcasecmp(name, (std::string(onpos ? vfilenames[onpos].shortname +
-		                                                  std::string(1, '\\')
-		                                        : "") +
-		                      cur_file->name)
-		                             .c_str()) == 0) {
-			stat_block->attr = (int)(cur_file->isdir
-			                                 ? DOS_ATTR_DIRECTORY
-			                                 : DOS_ATTR_ARCHIVE);
-			stat_block->size = cur_file->size;
-			stat_block->date = default_date;
-			stat_block->time = default_time;
-			return true;
-		}
-		cur_file = cur_file->next;
+	auto found = find_file_by_name(name);
+	if (found) {
+		stat_block->attr = (int)(found->isdir ? DOS_ATTR_DIRECTORY
+		                                      : DOS_ATTR_ARCHIVE);
+		stat_block->size = found->size;
+		stat_block->date = default_date;
+		stat_block->time = default_time;
+		return true;
 	}
 	return false;
 }
 
 bool Virtual_Drive::FileExists(const char* name){
 	assert(name);
-	auto cur_file = first_file;
-	while (cur_file) {
-		unsigned int onpos = cur_file->onpos;
-		if (strcasecmp(name, (std::string(onpos ? vfilenames[onpos].shortname +
-		                                                  std::string(1, '\\')
-		                                        : "") +
-		                      cur_file->name)
-		                             .c_str()) == 0)
-			return !cur_file->isdir;
-		cur_file = cur_file->next;
+	auto found = find_file_by_name(name);
+	if (found) {
+		return !found->isdir;
 	}
```

Comment:
Will change to:
```c++
if (vfile_name_exists(name)) {
    return !found->isdir;
}
```

Reply:
does not work because found IS the found file and the isdir info is needed